src/lib/settings/settings.py

```diff
@@ -0,0 +1,280 @@
+        return [
+            f"""{m_env["account_id"]}|{m_env["region"]}"""
+            for m_env in self._raw_settings[SettingFileNames.GENERAL].get(
+                "monitored_environments", []
+            )
+        ]
+
+    def get_delivery_method_names_raw_gs(self) -> List[str]:
+        """Retrieves the delivery_method names from the General settings (raw)."""
+        return [
+            dlvry_mthd["name"]
+            for dlvry_mthd in self._raw_settings[SettingFileNames.GENERAL].get(
+                "delivery_methods", []
+            )
+        ]
+
+    def get_monitoring_group_names_raw_mgs(self) -> List[str]:
+        """Retrieves the group_names from the Monitoring Groups settings (raw)."""
+        return [
+            m_grp["group_name"]
+            for m_grp in self._raw_settings[SettingFileNames.MONITORING_GROUPS].get(
+                "monitoring_groups", []
+            )
+        ]
+
+    def get_monitored_environment_names_raw_mgs(self) -> List[str]:
+        """Retrieves the monitored_environment_names from the Monitoring settings (raw)."""
+        resource_groups = []
+        for group in self._raw_settings[SettingFileNames.MONITORING_GROUPS].get(
+            "monitoring_groups", []
+        ):
+            for m_res in SettingConfigs.RESOURCE_TYPES:
+                resource_groups.extend(group.get(m_res, []))
+
+        return [res.get("monitored_environment_name") for res in resource_groups]
+
+    def get_monitoring_group_names_raw_rs(self) -> List[str]:
+        """Retrieves the monitoring_group names from the Recipients settings (raw)."""
+        return [
+            subscription["monitoring_group"]
+            for rec in self._raw_settings[SettingFileNames.RECIPIENTS].get(
+                "recipients", []
+            )
+            for subscription in rec.get("subscriptions", [])
+        ]
+
+    def get_delivery_method_names_raw_rs(self) -> List[str]:
+        """Retrieves the delivery_method names from the Recipients settings (raw)."""
+        return [
+            rec["delivery_method"]
+            for rec in self._raw_settings[SettingFileNames.RECIPIENTS].get(
+                "recipients", []
+            )
+        ]
+
+    # Factory methods
+    @classmethod
+    def from_file_path(cls, base_path: str) -> object:
+        general_settings = fm.read_file(
+            os.path.join(base_path, SettingFileNames.GENERAL)
+        )
+        monitoring_settings = fm.read_file(
+            os.path.join(base_path, SettingFileNames.MONITORING_GROUPS)
+        )
+        recipients_settings = fm.read_file(
+            os.path.join(base_path, SettingFileNames.RECIPIENTS)
+        )
+        return cls(general_settings, monitoring_settings, recipients_settings)
+
+    @classmethod
+    def from_s3_path(cls, base_path: str) -> object:
+        s3 = S3Manager()
+        general_settings = s3.read_file(
+            f"{base_path.rstrip('/')}/{SettingFileNames.GENERAL}"
```

Comment:
can we use os.path.join here as well?

>>> os.path.join('s3://mybucket/foo/bar', 'add/me')
's3://mybucket/foo/bar/add/me'
>>> os.path.join('s3://mybucket/foo/bar/', 'add/me')
's3://mybucket/foo/bar/add/me'
https://stackoverflow.com/questions/49078156/use-pathlib-for-s3-paths

Reply:
done